src/App/pages/ProductsPage/ProductsPage.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useRef, useState } from "react";
+
+import { Loader } from "@components/Loader";
+import { Product } from "@config/types";
+import axios from "axios";
+import InfiniteScroll from "react-infinite-scroll-component";
+
+import { Results } from "./components/Results";
+import { Search } from "./components/Search";
+import styles from "./ProductsPage.module.scss";
+
+export const ProductsPage = () => {
+  const [query, setQuery] = useState("");
+  const [productList, setProductList] = useState<Product[]>([]);
+  const dataChunk = useRef(40);
+
+  const fetchMore = async () => {
+    const res = await axios.get(
+      `https://api.escuelajs.co/api/v1/products?offset=${
+        dataChunk.current - 20
+      }&limit=20`
+    );
```

Comment:
не понял логику, почему в первый раз получаем продукты с 20-ого, можно использовать только `productList`, то есть
```suggestion
    const res = await axios.get(
      `https://api.escuelajs.co/api/v1/products?offset=${productList.length}&limit=20`
    );
```